caddyhttp/maxrequestbody/maxrequestbody.go

```diff
@@ -0,0 +1,59 @@
+package maxrequestbody
+
+import (
+	"regexp"
+	"strconv"
+
+	"github.com/mholt/caddy"
+	"github.com/mholt/caddy/caddyhttp/httpserver"
+)
+
+var SIZE_RE = regexp.MustCompile("[1-9][0-9]*[kmKM]?")
+
+func init() {
+	caddy.RegisterPlugin("maxrequestbody", caddy.Plugin{
+		ServerType: "http",
+		Action:     setupMaxRequestBody,
+	})
+}
+
+func setupMaxRequestBody(c *caddy.Controller) error {
+	var arg string
+	var path string = "/"
+
+	mult := int64(1)
+	config := httpserver.GetConfig(c)
+
+	if !c.Next() {
+		return c.ArgErr()
+	}
+
+	if !c.Args(&arg) {
+		return c.ArgErr()
+	}
+
+	if !SIZE_RE.MatchString(arg) {
+		// Optional path argument
+		path = arg
+
+		if !c.Args(&arg) {
+			return c.ArgErr()
+		}
+	}
+
+	// Required size argument
+	L := len(arg) - 1
+	if arg[L] == 'K' {
+		mult = 1024
+		arg = arg[0:L]
+	} else if arg[L] == 'M' {
+		mult = 1024 * 1024
+		arg = arg[0:L]
+	}
+	if size, err := strconv.ParseInt(arg, 10, 64); err != nil {
+		return c.ArgErr()
+	} else {
+		config.MaxRequestBodySizes = append(config.MaxRequestBodySizes, httpserver.PathLimit{path, size * mult})
```

Comment:
Hmm, I thought there was a facility in the `httpserver` package to help with the longest-path-matching thing, but maybe it was only the `proxy` middleware that got that blessed feature. (Man, I still really think we need to totally standardize the code to match requests. But that's a different topic.) Check out the proxy middleware maybe...